### PR TITLE
Black out cells

### LINKED ISSUE #26: Black out cells

### PR DESCRIPTION
**Pull request by 0beyer**
When a cell has all edges bolded, the cell becomes black.
If you hold ctrl and click on a cell, all edges get bolded and the cell gets blacked out.
If you ctrl click on a blacked out cell, all edges get un-bolded and the cell is un-blacked out.

We also created toggle edge, which checks an adjacent cells edge, and toggles both the current and adjacent edge to either true or false.

### CODE REVIEW

File: client/cypress/e2e/grid-component.cy.ts
Comment (Bonesbhbhbh) [inline client/cypress/e2e/grid-component.cy.ts:1]: Non-critical but why are you checking that mat-grid-title is correct by length and not the actual title? Seems like unnecessarily lowering the quality of the test

File: client/cypress/e2e/grid-component.cy.ts [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
+  it('should render the grid with default size', () => {
+    cy.get('app-grid-component').within(() => {
+      cy.get('mat-grid-tile').should('have.length', 100);
     });
-  });
\ No newline at end of file
+  });
+
+  it('should black-out the cell with all edges bolded', () => {
+    cy.get('app-grid-cell').first().click().type('{ctrl}{rightarrow}{downarrow}{uparrow}{leftarrow}');
```
Comment (Bonesbhbhbh) [inline client/cypress/e2e/grid-component.cy.ts]: This test does nothing besides checking that app-grid-cell exists and is able to be typed into. Please finish writing this test or rename it to reflect what you are trying to check

File: client/src/app/grid-cell/grid-cell.component.spec.ts [hunk 2]
```diff
@@ -34,21 +34,84 @@ describe('GridCellComponent', () => {
...
+
+    const cellElement: HTMLElement = fixture.nativeElement.querySelector('.cell');
+    expect(component.gridCell.edges.top).toBeTrue();
+    expect(component.gridCell.edges.left).toBeTrue();
+    expect(component.gridCell.edges.right).toBeTrue();
+    expect(component.gridCell.edges.bottom).toBeTrue();
+    expect(cellElement.classList).toContain('bold-top');
+    expect(cellElement.classList).toContain('bold-right');
+    expect(cellElement.classList).toContain('bold-bottom');
+    expect(cellElement.classList).toContain('bold-left');
```
Comment (Bonesbhbhbh) [inline client/src/app/grid-cell/grid-cell.component.spec.ts:86]: Should you also be checking if the background is blacked out?

File: client/src/app/grid-cell/grid-cell.component.spec.ts [hunk 3]
```diff
@@ -34,21 +34,84 @@ describe('GridCellComponent', () => {
...
+
+    const cellElement: HTMLElement = fixture.nativeElement.querySelector('.cell');
+    expect(component.gridCell.edges.top).toBeFalse();
+    expect(component.gridCell.edges.left).toBeFalse();
+    expect(component.gridCell.edges.right).toBeFalse();
+    expect(component.gridCell.edges.bottom).toBeFalse();
+    expect(cellElement.classList).not.toContain('bold-top');
+    expect(cellElement.classList).not.toContain('bold-right');
+    expect(cellElement.classList).not.toContain('bold-bottom');
+    expect(cellElement.classList).not.toContain('bold-left');
```
Comment (Bonesbhbhbh) [inline client/src/app/grid-cell/grid-cell.component.spec.ts:118]: see comment in "should bold after ctrl click" test

File: client/src/app/grid-cell/grid-cell.component.ts [hunk 4]
```diff
@@ -79,37 +82,78 @@ export class GridCellComponent {
     this.gridCell.editable = state;
   }
 
+  toggleEdge(edge: string) {
```
Comment (Bonesbhbhbh) [inline client/src/app/grid-cell/grid-cell.component.ts:85]: consider testing more of the cases involved in this method

File: client/cypress/e2e/grid-component.cy.ts
Comment (0beyer) [reply client/cypress/e2e/grid-component.cy.ts:1]: its mat-grid-tile not title, it checks the amount of cells

File: client/src/app/grid-cell/grid-cell.component.ts (hunk 4 above)
Comment (0beyer) [reply client/src/app/grid-cell/grid-cell.component.ts:85]: the parts of toggle edge that is not tested was the stuff i was not able to work out yesterdat

File: client/src/app/grid-cell/grid-cell.component.spec.ts (hunk 3 above)
Comment (0beyer) [reply client/src/app/grid-cell/grid-cell.component.spec.ts:118]: see previous

File: client/src/app/grid-cell/grid-cell.component.spec.ts (hunk 2 above)
Comment (0beyer) [reply client/src/app/grid-cell/grid-cell.component.spec.ts:86]: added a check for this, thanks for noticing my mistake

File: client/cypress/e2e/grid-component.cy.ts (hunk 1 above)
Comment (0beyer) [reply client/cypress/e2e/grid-component.cy.ts]: this test actually does sompthing now, i will update my pull request now

**Review by Bonesbhbhbh — APPROVED** [review]
Looks good 👍

**Pull request merged**